vyxal/elements.py

```diff
@@ -936,16 +936,16 @@ def expe_minus_1(lhs, ctx):
 def exponent(lhs, rhs, ctx):
     """Element e
     (num, num) -> a ** b (exponentiation)
-    (str, num) -> every bth character of a
-    (num, str) -> every ath character of b
+    (num, str) -> repeat characters of b until b is length a
+    (str, num) -> repeat characters of a until a is length b
     (str, str) -> regex.search(pattern=a, string=b).span() (Length of regex match)
     """
 
     ts = vy_type(lhs, rhs)
     return {
         (NUMBER_TYPE, NUMBER_TYPE): lambda: lhs ** rhs,
-        (NUMBER_TYPE, str): lambda: rhs[:: int(lhs)],
-        (str, NUMBER_TYPE): lambda: lhs[:: int(rhs)],
+        (NUMBER_TYPE, str): lambda: (rhs * lhs)[:lhs],
```

Comment:
Multiplying by the number will get a pretty long string (and won't work for negative numbers). How about only getting as many whole strings as you need and then concatenating a slice?